### ISSUE TITLE
Unable to create backup on Sony Xperia Z5 Compact despite chat backups being enabled

### ISSUE THREAD

**Issue by oXvTV** [signalapp/Signal-Android#9083]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Unable to create backup on Sony Xperia Z5 Compact despite chat backups being enabled

I have recently obtained a new phone and I would like to transfer my messages from Signal on my old phone to my new phone. Obviously, in order to do this, the first thing I need to do is create a backup on Signal on my old phone. Unfortunately, for some unknown reason, I am unable to create a backup on my Sony Xperia Z5 Compact running Android version 7.1.1 - when I press ‘Create backup’ nothing happens. This is despite the fact that I have enabled ‘Chat backups’, ensured the necessary permissions for Signal are enabled, ensured there is enough space to create a backup on the device, and I do not have an SD card in my device. Of course this is very frustrating as I would like to transfer my messages to my new phone so I can begin to use it. I have contacted the official email support but have not had a response yet. I would be very grateful for your help.

### Steps to reproduce
- open Signal and settings
- enable chat backups
- try to create backup by pressing 'Create backup'

**Actual result:** 
Nothing happens after running the steps above except that a Signal notification symbol very briefly appears in the notification panel at the top of the screen for a few milliseconds or so and then disappears.
**Expected result:** 
I'm not exactly sure what the expected behaviour should be but I imagine a backup should be created somehow.

### Device info

**Device:** Sony Xperia Z5 Compact
**Android version:** 7.1.1
**Signal version:** 4.47.7

### Link to debug log

https://debuglogs.org/ec45c190c0874f8987621635d0026ec522625f65d2007196fb14bdf41eeade71

**Comment by oXvTV** [comment]
Could someone please help me with this? I have tried to seek help through every possible channel I have found but with no luck. Help would be much appreciated so I can get this sorted, an important issue for me! Thank you

**Comment by Cakesake** [comment]
My issue might be the same as this. The debug log link isn't working for me but here's mine:
https://debuglogs.org/b60b7d7e6f41095e6c9a916fb854d50d67081faebc9593172b56c65a88f9f641

My issue seems similar to this one  
https://github.com/signalapp/Signal-Android/issues/8054
with the NoExternalStorageException thrown by 
```
    File storage = Environment.getExternalStorageDirectory();

    if (!storage.canWrite()) {
      throw new NoExternalStorageException();
    }
```
However, unlike that issue, I'm not running Lineage. I recently changed phones, and Whatsapp was able to create a backup folder in External Storage in the same way that org.thoughtcrime.securesms.util.StorageUtil.getBackupDirectory() looks designed to do.

I'm not an Android developer so I'm not sure if this is easy to check, but could the issue be that Signal isn't handling runtime permissions? ( https://stackoverflow.com/a/36310497 ) I can see in the log that I have the WRITE_EXTERNAL_STORAGE permission.

**Comment by MichaelMure** [comment]
I have a similar issue on a OnePlus2 running Lineage 14.1, Android 7.1.2. There is no external storage (no SD card slot), enough space and all permissions are enabled. So far I managed to find that:
- I can enable the chat backups but no backup is created, even after multiple days.
- enabling the backups create an background alarm ~17 minutes in the future: `PersistentAlarmManagerListener: class org.thoughtcrime.securesms.service.LocalBackupListener scheduling for: 1575378266977`
- the phone clock is automatically synced with the network and correctly set
- when this alarm should trigger, nothing happen. No log, nothing.
- triggering manually a backup with the "Create backup" button fail with `JobController: [c80d08da-cf62-489f-b1de-be20b81e9ec9][LocalBackupJob] Already at the max instance count of 1. Skipping. (Time Since Submission: 5 ms, Lifespan: Immortal, Run Attempt: 1/3)`

**Comment by oXvTV** [comment]
Can someone please help me with this? It’s been a few months now and I haven’t been able to get any help after having tried to reach out to all the channels I can. I’ve tried fixing the problem again after updating the app but the problem still persists and is the same. Here is a new logfile for the problem occurring with the new app:

https://debuglogs.org/8b7508e70c8df7f703b29c25a536003c6f4f06d8f65045ff4b38558d4a983fa0

I would really appreciate some help as I haven’t been able to use Signal for months on my new phone because of this.

**Comment by oXvTV** [comment]
I actually managed to get this working. Because my phone screen was broken and I couldn't use the touch screen to operate the phone I was having to operate it connecting the phone to a computer using a USB cable and Vysor software. Whenever I tried to backup through the app using Vysor, it didn't worked (this is the method I was trying for months). I then thought that having my phone connected to my computer via USB somehow was preventing backup. So I then tried the same thing using TeamViewer and, amazingly, this method worked. I'm not sure what difference this would have made but it seemed to work in any case. I was pretty relieved and happy when it eventually worked :-) I don't know why I forgot to mention this detail about my phone screen not working and having to operate it through other means. Perhaps someone would have realised what the problem was sooner if I had.
